### PR TITLE
Make `threaded` a property of Context

### PR DESCRIPTION
**Pull request by zbanks**
* Context, at creation, is set to either threaded (e.g. for GUI) or unthreaded (e.g. for CLI)
* When new `OpenGLWorkerContext`s are created, they use this global threadedness value

### CODE REVIEW

File: src/SelfTimedReadBackOutputNode.cpp [hunk 1]
```diff
@@ -157,10 +159,11 @@ void STRBONOpenGLWorker::stop() {
         qWarning() << "Node not ready, ignoring start";
```
Comment (ervanalb) [inline src/SelfTimedReadBackOutputNode.cpp]: can you also fix this string

File: src/SelfTimedReadBackOutputNode.cpp [hunk 2]
```diff
@@ -120,6 +121,7 @@ QSharedPointer<QOpenGLShaderProgram> STRBONOpenGLWorker::loadBlitShader() {
 
 void STRBONOpenGLWorker::initialize(QSize size) {
     makeCurrent();
+    Q_ASSERT(QThread::currentThread() == thread());
```
Comment (ervanalb) [inline src/SelfTimedReadBackOutputNode.cpp]: I think this should be one line up?

File: src/SelfTimedReadBackOutputNode.cpp [hunk 3]
```diff
@@ -157,10 +159,11 @@ void STRBONOpenGLWorker::stop() {
         qWarning() << "Node not ready, ignoring start";
         return;
     }
-    m_timer->start();
+    m_timer->stop();
 }
 
 void STRBONOpenGLWorker::onTimeout() {
+    Q_ASSERT(QThread::currentThread() == thread());
```
Comment (ervanalb) [inline src/SelfTimedReadBackOutputNode.cpp]: Can you add `Q_ASSERT` to all the slots

**Comment by zbanks** [comment]
Added requested changes.

File: src/SelfTimedReadBackOutputNode.cpp [hunk 4]
```diff
@@ -40,21 +41,25 @@ SelfTimedReadBackOutputNode::SelfTimedReadBackOutputNode(QSharedPointer<SelfTime
 }
 
 void SelfTimedReadBackOutputNode::start() {
+    Q_ASSERT(QThread::currentThread() == thread());
```
Comment (ervanalb) [inline src/SelfTimedReadBackOutputNode.cpp]: wait these don't need `Q_ASSERT`, they are thread-safe

**Pull request merged**